src/infrastructure/adapters/PluginRegistry.ts

```
@@ -0,0 +1,517 @@
+
+	/**
+     * Emit plugin event
+     */
+    emit(eventType: string, event: PluginEvent): void {
+        const handlers = this.eventBus.get(eventType) || []
+        handlers.forEach(handler => {
+            try {
+                handler(event)
+            } catch (error) {
+                Effect.logError(`❌ Event handler error for ${eventType}: ${error}`)
+            }
```

ESLint

[error] 🚨 BANNED: try/catch blocks. Use Effect.gen() with proper error handling via Effect.catchAll() or Effect.orElse().

(no-restricted-syntax)